Firestore/Example/Tests/Integration/FSTStreamTests.m

```diff
@@ -38,17 +38,25 @@ - (void)writesFinishedWithError:(NSError *_Nullable)error;
  */
 @interface FSTStreamStatusDelegate : NSObject <FSTWatchStreamDelegate, FSTWriteStreamDelegate>
 
+- (instancetype)initFrom:(XCTestCase *)testCase
+              usingQueue:(FSTDispatchQueue *)dispatchQueue NS_DESIGNATED_INITIALIZER;
+- (instancetype)init NS_UNAVAILABLE;
+
 @property(nonatomic, readonly) NSMutableArray<NSString *> *states;
 @property(atomic, readwrite) BOOL invokeCallbacks;
 @property(nonatomic, weak) XCTestExpectation *expectation;
 @property(nonatomic, weak, readonly) FSTStream *stream;
+@property(nonatomic, weak, readonly) XCTestCase *testCase;
+@property(nonatomic, weak, readonly) FSTDispatchQueue *dispatchQueue;
```

Comment:
There are a lot of weak references in here which is a serious design smell. Are these actually necessary?

In general avoid weak references unless you specifically need them to break a reference cycle because they make your code very difficult to reason about.

In this case, you need the `testCase` to be weak because you're making this a member of that test case (which creates a cycle). The other things in here can be regular strong references.

Reply:
It's enough for 'testCase' to be weak. Everything else is only retained by the test case itself as well as 'FSTStreamStatusDelegate'.